wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -247,17 +249,18 @@ markWalletReady cid isReady = do
 -- Deleters
 ----------------------------------------------------------------------------
 
-deleteWallet :: MonadWalletLogic ctx m => CId Wal -> m ()
+deleteWallet :: MonadWalletLogic ctx m => CId Wal -> m NoContent
 deleteWallet wid = do
     accounts <- getAccounts (Just wid)
     mapM_ (deleteAccount <=< decodeCTypeOrFail . caId) accounts
     removeWallet wid
     removeTxMetas wid
     removeHistoryCache wid
     deleteSecretKey . fromIntegral =<< getAddrIdx wid
+    return NoContent
 
-deleteAccount :: MonadWalletLogic ctx m => AccountId -> m ()
-deleteAccount = removeAccount
+deleteAccount :: MonadWalletLogic ctx m => AccountId -> m NoContent
+deleteAccount aid = removeAccount aid >> return NoContent
```

Comment:
Since you go with one-liner, why not `removeAccount aid $> NoContent`

Reply:
No particular reason and I don't have a strong opinion on whether or not we should use one or the other, so I'm happy to go with the applicative style here.